crates/rpc/src/server/accept.rs

```diff
@@ -1,61 +1,146 @@
 use std::task::{Context as StdContext, Poll};
 
 use futures::{FutureExt, future::BoxFuture};
-use http::header::ACCEPT;
+use http::{
+    HeaderMap, HeaderValue,
+    header::{ACCEPT, ToStrError},
+};
+use miden_node_utils::ErrorReport;
+use miden_objects::Digest;
+use miden_tx::utils::HexParseError;
 use nom::bytes::complete::{tag, take_until};
 use semver::{Version, VersionReq};
 use tower::{Layer, Service};
-use tracing::debug;
 
-use crate::COMPONENT;
-
-/// Layer responsible for handling HTTP ACCEPT headers.
+/// Rejects requests whose headers indicate a mismatch in RPC version or the network.
+///
+/// The former is taken from the `accept` header with the expected format
+/// `application/vnd.miden.<major>.<minor>.<patch>+grpc`.
+///
+/// The latter expects the network's genesis block's commitment in the `miden-network` header.
```

Comment:
Wondering if we should just re-use the accept header for this. If we plan on putting more things like this in place, I don't think we should. But if this is the only other verification for client/server/network matching, then we could integrate it e.g.
```
application/vnd.miden.<network>.<major>.<minor>.<patch>+grpc

Reply:
> So having two separate "media type" as part of the header would be incorrect afaict
Yep.

> q is intended to let the user set priorities, a higher value means they prefer this content type.

Is this to say that `; network=<..>` would be inappropriate?

Reply:
Is `network` self-explanatory enough here? maybe it should be `genesis=0x123...`? Though, probably doesn't matter too much.

Reply:
Yea network is a bit misleading. Genesis is better.

Reply:
Yes I think that is practical for us and also idiomatic as far as I could tell from the relevant RFCs.